packages/experimental/assistant/src/conversation/conversation.ts

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { invariant } from '@dxos/invariant';
+import { log } from '@dxos/log';
+
+import type { AIBackend } from './backend/interface';
+import { type LLMMessage, type LLMModel, type LLMTool } from './types';
+
+export type CreateLLMConversationParams = {
+  model: LLMModel;
+  messages: LLMMessage[];
+
+  /**
+   * System prompt that specifies instructions for the LLM.
+   */
+  system?: string;
+
+  tools: LLMTool[];
+  backend: AIBackend;
+
+  logger?: (event: ConversationEvent) => void;
+};
+
+export type ConversationEvent = {
+  type: 'message';
+  message: LLMMessage;
+};
+
+export const runLLM = async (params: CreateLLMConversationParams) => {
```

Comment:
while i understand this is early, i'm afraid of naked functions of this potential complexity. what is the appropriate model of a conversation?